### ISSUE TITLE
Inverted Buffer in FG

### ISSUE THREAD

**Issue by matfax** [cdozdil/OptiScaler#99]
Some buffers are inverted, as shown in this debug view. The upper left, lower left, and upper center views are vertically inverted (y values) in CSL2.

![Screenshot](https://github.com/user-attachments/assets/ad8adaf6-9d08-468c-9007-5fc0851c3862)

I think it could be resolved by adding buffer invert options similar to those that Reshade uses.

**Comment by cdozdil** [comment]
Oh, this is interesting. Could you provide a log with `LogLevel=0` set at `nvngx.ini`?

**Comment by matfax** [comment]
Here you go.

[OptiScaler.log](https://github.com/user-attachments/files/17588181/OptiScaler.log)

**Comment by cdozdil** [comment]
Thanks, looks like this is a classical Unity situation. Will check what can I do 👍

**Comment by matfax** [comment]
Oddly enough, upscaling debug view doesn't have the issue.

![Screenshot 2024-11-12 200924](https://github.com/user-attachments/assets/f31ce5e7-33fe-4c74-8cac-ae78df0659fc)

Like the flag is only passed to the upscaler lib but not the frame gen lib.

**Comment by cdozdil** [comment]
Actually that's upside down too, that red depth view should be at top right.

**Comment by matfax** [comment]
Here are two better screenshots.

![Screenshot 2024-11-13 195724](https://github.com/user-attachments/assets/1c8c0831-aae0-4b01-a8a7-6f623aaf5e7b)
![Screenshot 2024-11-13 195715](https://github.com/user-attachments/assets/b75f41c1-7df4-4253-88f6-022971341605)

I'm not sure about the debug view "view order", if that's what you mean, but the views themselves seem to be correctly oriented.

**Comment by cdozdil** [comment]
Check image below, this is normal view of debug view.

![image](https://github.com/user-attachments/assets/94f7b334-3823-4540-9a68-925d64e1b332)

Gray box on bottom left is normally at upper left. So image upscaled upside down and Unity flipped it.

**Comment by Gonckzilla** [comment]
Took me a while to realize exactly what was going on using Optiscaler on Lego Builder's Journey (Unity 😠). Yeah it's upside down, and depth is not inverted apparently, I can move a brick and see the "FG ghost" moving around, it's mirrored when "depth inverted" is on.

**Comment by matfax** [comment]
Yes, as described - it's not a z axis issue, so it can't be fixed by the depth buffer inversion option that targets this axis. The "inversion" exists on the horizontal axis. It's not just a debug view issue but it's fed to the FG in a mirrored state.

**Comment by TheRazerMD** [comment]
New Nightly build was released with an FG flip for Unity option, so you can try it now to see if it helped.

https://github.com/optiscaler/OptiScaler/releases/tag/nightly

**Comment by matfax** [comment]
I haven't managed to test the nightly feature yet because CS2 doesn't run in DX12 mode anymore without crashing, at least on my system. It runs in DX11 though, without FG. Nvidia SM also causes crashes, unless SM Feature Flag 0x4 is set in the profile. Then, SM works according to the monitor framerate, but it doesn't show it as enabled in the Nvidia Stat Overlay.
